riscv/src/riscv_interpreter.rs

```diff
@@ -54,7 +54,19 @@ pub fn riscv_interpreter(code: &[u32]) -> Vec<RiscvInstruction> {
 
         // Ignore instructions that are zero
         if inst == 0 {
-            //println!("riscv_interpreter() found inst=0 at position s={}", s);
+            println!("riscv_interpreter() found inst=0 at position s={} (index in u32 array)", s);
+            // Create an addi x0, x0, 0 instruction (standard RISC-V NOP)
+            let i = RiscvInstruction {
+                rvinst: inst,
+                t: "I".to_string(),
+                inst: "addi".to_string(),
+                rd: 0,  // x0 destination
+                rs1: 0, // x0 source
+                rs2: 0,
+                imm: 0, // immediate value 0
+                ..Default::default()
+            };
+            insts.push(i);
```

Comment:
I am yet to create a good test for this because that fails loudly; the rom-setup and emulator will still pass.

If you run this over the go-program-elf, notice that the emulator exits early with less steps and does not print out hello world, if this is not here. While if this is here, the emulator will run for approximately 2M steps and print out hello world.